hq/app/schedule/IamAudit.scala

```diff
@@ -0,0 +1,64 @@
+package schedule
+
+import com.gu.anghammarad.models.{Notification, AwsAccount => Account}
+import logic.DateUtils
+import model._
+import schedule.IamMessages.createMessage
+import schedule.IamNotifier.createNotification
+import utils.attempt.FailedAttempt
+
+object IamAudit {
+  def makeCredentialsNotification(allCreds: Map[AwsAccount, Either[FailedAttempt, CredentialReportDisplay]]):List[Either[FailedAttempt, Notification]] = {
+    allCreds.toList.map { case (awsAccount, eFCreds) =>
+      eFCreds.map { credsReport =>
+        val outdatedKeys = outdatedKeysInfo(findOldAccessKeys(credsReport))
+        val missingMfa = missingMfaInfo(findMissingMfa(credsReport))
+        val message = createMessage(outdatedKeys, missingMfa)
+        createNotification(Account(awsAccount.id), message)
+      }
+    }
+  }
+
+  def findOldAccessKeys(credsReport: CredentialReportDisplay): CredentialReportDisplay = {
+    val filteredMachines = credsReport.machineUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    val filteredHumans = credsReport.humanUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    credsReport.copy(machineUsers = filteredMachines, humanUsers = filteredHumans)
+  }
+
+  def hasOutdatedKey(keys: List[AccessKey]): Boolean = keys.exists(key => DateUtils.dayDiff(key.lastRotated).getOrElse(1L) > 90L)
```

Comment:
I'm needing to add `L` here, otherwise I get a compile error that the compiler doesn't see these values as Longs. I wonder if this can be improved 😊

Reply:
I think the 90L should go in the Config.scala object as 'recommendedRotationCadence' or similar. The L is fine, though you might not need it if you do `val blah: Long = 90`

Also, we have been recommending to teams to rotate machine credentials once a year and human credentials once every 3 months so there might be more than one concept of outdated

Reply:
Ah fantastic, I wasn't aware of this, that's great! I'll make that change 👍🏽